### DISCUSSION TITLE
Add Enum Support To Table Editor

### DISCUSSION THREAD

**Discussion by dweekly** [supabase/supabase#11703] [Feature Requests]
# Update - Feature shipped: 
We've shipped this feature. You can find it [in the Dashboard](https://supabase.com/dashboard/project/_/database/types). Here's the [PR](https://github.com/supabase/supabase/pull/17953). Please [let us know](https://github.com/orgs/supabase/discussions/categories/general) if you have any feedback! 

---

# Feature request

Supabase has supported Postgres ENUM types for two years but the web UI (Table Editor) still does not yet appear to support creating columns with ENUM type. In order to do this, it looks like I'll need to modify the table manually after creating it. Given the prevalence of ENUM columns in table design, it would seem desirable to have support for creating ENUM columns in the web UI.

## Describe the solution you'd like

In the table creation wizard, the "Type" pulldown for a column should include an "Enum" option. A popup could then collect a list of allowable values from the user (or in a simpler first implementation, the user could be asked to comma separate the list of possible values in a TEXTAREA).

## Describe alternatives you've considered

The UI could expose a "..." next to types that require configuration, not just ENUM but also possibly VARCHAR. (It's not clear how to set character limits for a VARCHAR in the UI today).

## Additional context

ENUMs have been supported in Postgres for well over a decade and are not considered novel. Their use is widespread in database design; it would be very nice to have them supported as a first class citizen in the Table Editor UI.

**Comment by ChuckJonas** [comment]
This seems like a big "quality of life" improvement with what should be minimal effort.  

IMO, creating new enum's in the table editor isn't absolutely necessary.  All that's really needed is just place in the database tab to see, edit and create new enums:

<img width="249" alt="Default_Project___Default_Organization___Supabase" src="https://github.com/supabase/supabase/assets/5217568/ee4fca4f-446a-46df-a306-0238d97aea23">

**Comment by stolinski** [comment]
This is something I'd really like to see. Some feedback overall that because the docs tend to only show the SQL Editor and the web UI is missing things like enums in the table editor, it makes the UI of the table editor feel a bit like a 2nd class citizen in the Supabase world, which is a shame because the UI for the table editor is so great.

I end up opting for a lazy solution like using a text field instead of an enum just because the table editor is fast.

**Comment by Brentably** [comment]
yeah this would be lit

**Comment by MildTomato** [comment]
@joshenlim has been working on this topic

https://github.com/supabase/supabase/pull/17953

**Comment by saltcod** [comment]
Thanks all for the request. This feature went out today. Feedback appreciated!

**Comment by Brentably** [comment]
Awesome! Thanks

**Comment by MildTomato** [comment]
# Update - Feature shipped: 
We've shipped this feature. You can find it [in the Dashboard](https://supabase.com/dashboard/project/_/database/types). Here's the [PR](https://github.com/supabase/supabase/pull/17953). Please [let us know](https://github.com/orgs/supabase/discussions/categories/general) if you have any feedback!

**Reply by EskelCz** [reply]
@MildTomato  I love it, thanks a lot. <3 However, the **table editor** doesn't allow to change the values in the columns that use ENUMs. Should I open a new feature request for it?
